Formula/tfel.rb

```diff
@@ -0,0 +1,69 @@
+class Tfel < Formula
+  desc "Main repository of TFEL/MFront project"
+  homepage "https://thelfer.github.io/tfel/web/index.html"
+  license "GPL-1.0-or-later"
+  head "https://github.com/thelfer/tfel.git", using: :git, branch: "master"
+  stable do
+    url "https://github.com/thelfer/tfel/archive/refs/tags/TFEL-4.0.tar.gz"
+    sha256 "ed3fb2f59c6b8c9896606ef92276f81942433dd5f60d8130ba07c3af80b039e2"
+  end
+
+  depends_on "cmake" => :build
+  depends_on "gcc" => :required
+  depends_on "python@3.9" => :required
+
+  on_linux do
+    fails_with gcc: "5"
+  end
+
+  def install
+    mkdir "build" do
+      build_type = "-DCMAKE_BUILD_TYPE=Release"
+      args = ["-Denable-reference-doc=ON", "-Denable-website=ON"]
+      args << "-Dlocal-castem-header=ON"
+      args << "-Denable-fortran=OFF"
+      args << "-Denable-python=ON"
+      args << "-Denable-python-bindings=OFF"
+      args << "-Denable-cyrano=ON"
+      args << "-Denable-lsdyna=ON"
+      args << "-Denable-aster=ON"
+      args << "-Denable-abaqus=ON"
+      args << "-Denable-calculix=ON"
+      args << "-Denable-comsol=ON"
+      args << "-Denable-diana-fea=ON"
+      args << "-Denable-ansys=ON"
+      args << "-Denable-europlexus=ON"
+      args << "-Denable-portable-build=ON"
```

Comment:
What does this do?

Reply:
So... what does it do?

Reply:
These are the cmake compilation options for interfacing tfel/mfront with different software. I added a comment in the .rb file to clarify it.

Reply:
And what does the line I commented on `args << "-Denable-portable-build=ON"` do? because that does not seem like an interface option

Reply:
Indeed, by default, the compilation of `mfront` takes advantage of the
specific CPU of the host system. However,
the binaries generated can not:

- be used to generated redistributable binary packages.
- be installed on a NFS-like shared folder to be shared on a network
  of computers.

In both cases, the execution can fail with an `illegal instruction`
message.

To solve this issue, the `-Denable-portable-build=ON` option has been introduced.

With this option, the `mfront` binary and all the `TFEL` shared
libraries will be "portable" (will not include CPU specific
instructions).

Reply:
Isn't that already covered by Homebrew's build system suggesting an older architecture to the compiler?

Reply:
It depends if homebrew specifies appropriate compiler flags when building packages, like anaconda does. Does homebrew sets CFLAGS, CXXFLAGS and FFLAGS ?

If so, we can use another cmake option telling `TFEL` to use those variables rather than the one its selects by default (-DUSE_EXTERNAL_COMPILER_FLAGS=ON).

Otherwise, we can let `TFEL` selects its compiler flags, but we need to skip architecture dependent flags and thus use the `-Denable-portable-build=ON` option.

Reply:
I guess that depends on what you consider "appropriate compiler flags" but seeing how they work for all other software, I guess it does.

Reply:
I made a small test and added `system "bash", "-c", "echo $CXXFLAGS"` to my formula. It seems that `CXXFLAGS` is empty. So my best choice here is to use `-Denable-portable-build=ON` which basically sets `CXXFLAGS` to `-O2`. Would you agree ?

Reply:
Well, bash doesn't need those so we don't set it there. We set them in our compiler wrapper.